performance/vip-tweaks.php

```diff
@@ -80,3 +80,77 @@ function wpcom_vip_media_library_months_with_files() {
 		return $months;
 	}
 }
+
+/**
+ * Improves performance of all the post listing pages that display the months dropdown filter.
+ * A unique transient is created for each post type.
+ */
+if ( is_admin() ) {
+
+	add_filter( 'pre_months_dropdown_query', 'wpcom_vip_pre_months_dropdown_query', 10, 2 );
+	function wpcom_vip_pre_months_dropdown_query( $months, $post_type ) {
```

Comment:
I wonder if it's worth refactoring this as a generic function for the dropdown functions in wp-admin - the logic and queries are identical as with `wpcom_vip_media_library_months_with_files`